### PR TITLE
Add _status action

### PR DESCRIPTION
**Pull request by mmihic**
Along with a DefaultActionsModule which installs support for all
default actions (_readiness, _liveness, _metrics, _status + not found
handling)

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/actions/DefaultActionsModule.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import misk.inject.KAbstractModule
+import misk.metrics.web.MetricsJsonAction
+import misk.web.WebActionModule
+
+/**
+ * Installs default actions for handling errors, providing status and readiness, and handling
+ * resources that aren't found
+ */
+class DefaultActionsModule : KAbstractModule() {
```
Comment (swankjesse) [inline misk/src/main/kotlin/misk/web/actions/DefaultActionsModule.kt]: should this come by default when you install a JettyModule?

File: misk/src/main/kotlin/misk/web/actions/LivenessCheckAction.kt [hunk 2]
```diff
@@ -10,19 +10,18 @@ import javax.inject.Inject
...
+) : WebAction {
 
   @Get("/_liveness")
   @ResponseContentType(MediaTypes.TEXT_PLAIN_UTF8)
   fun livenessCheck(): Response<String> {
     val isAlive = services.all {
       it.state() != State.FAILED && it.state() != State.TERMINATED
     }
+
     // TODO(jgulbronson) - Should return an empty body
```
Comment (swankjesse) [inline misk/src/main/kotlin/misk/web/actions/LivenessCheckAction.kt]: TODO done?

Comment (swankjesse) [reply misk/src/main/kotlin/misk/web/actions/LivenessCheckAction.kt]: Alternately, we could just make a ResponseBody.EMPTY constant that has no content-type and is 0 bytes long

File: misk/src/main/kotlin/misk/web/actions/StatusAction.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  @Get("/_status")
+  @ResponseContentType(MediaTypes.APPLICATION_JSON)
+  fun getStatus(): ServerStatus {
+    val serviceStatus = services.map { it.javaClass.simpleName to it.state() }.toMap()
+    val healthCheckStatus = healthChecks.map { it.javaClass.simpleName to it.status() }.toMap()
+    return ServerStatus(serviceStatus, healthCheckStatus)
+  }
+
+  data class ServerStatus(
```
Comment (swankjesse) [inline misk/src/main/kotlin/misk/web/actions/StatusAction.kt]: one more thing for me to be annoying about – does Misk wanna do camelCase or snake_case for stuff that ends up on JSON?

Comment (mmihic) [reply misk/src/main/kotlin/misk/web/actions/StatusAction.kt]: Personally I prefer snake_case for JSON

Comment (eeSeeGee) [reply misk/src/main/kotlin/misk/web/actions/StatusAction.kt]: Me too, but should it be converted during the marshall step?

Comment (mmihic) [reply misk/src/main/kotlin/misk/web/actions/StatusAction.kt]: https://publicobject.com/2016/01/20/strict-naming-conventions-are-a-liability/

**Pull request merged**